org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterThermostatSystemMode.java

```diff
@@ -0,0 +1,124 @@
+ *
+ */
+public class ZigBeeConverterThermostatSystemMode extends ZigBeeBaseChannelConverter implements ZclAttributeListener {
+    private Logger logger = LoggerFactory.getLogger(ZigBeeConverterThermostatSystemMode.class);
+
+    private ZclThermostatCluster cluster;
+
+    @Override
+    public boolean initializeConverter() {
+        cluster = (ZclThermostatCluster) endpoint.getInputCluster(ZclThermostatCluster.CLUSTER_ID);
+        if (cluster == null) {
+            logger.error("{}: Error opening device thermostat cluster", endpoint.getIeeeAddress());
+            return false;
+        }
+
+        CommandResult bindResponse;
+        try {
+            bindResponse = bind(cluster).get();
+            if (!bindResponse.isSuccess()) {
+                logger.debug("{}: Failed to bind thermostat cluster", endpoint.getIeeeAddress());
+            } else {
+                // Configure reporting
+                ZclAttribute attribute = cluster.getAttribute(ZclThermostatCluster.ATTR_SYSTEMMODE);
+                cluster.setReporting(attribute, REPORTING_PERIOD_DEFAULT_MIN, REPORTING_PERIOD_DEFAULT_MAX, 0.1);
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("{}: Exception setting reporting ", endpoint.getIeeeAddress(), e);
+        }
+
+        // Add a listener, then request the status
+        cluster.addAttributeListener(this);
+
+        return true;
+    }
+
+    @Override
+    public void disposeConverter() {
+        cluster.removeAttributeListener(this);
+    }
+
+    @Override
+    public void handleRefresh() {
+        cluster.getLocalTemperature(0);
+    }
+
+    @Override
+    public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
+        ZclThermostatCluster cluster = (ZclThermostatCluster) endpoint.getInputCluster(ZclThermostatCluster.CLUSTER_ID);
+        if (cluster == null) {
+            logger.trace("{}: Thermostat cluster not found", endpoint.getIeeeAddress());
+            return null;
+        }
+
+        try {
+            if (!cluster.discoverAttributes(false).get()) {
+                // Device is not supporting attribute reporting - instead, just read the attributes
+                Integer capabilities = cluster.getSystemMode(Long.MAX_VALUE);
+                if (capabilities == null) {
+                    logger.trace("{}: Thermostat system mode returned null", endpoint.getIeeeAddress());
+                    return null;
+                }
+            } else if (!cluster.isAttributeSupported(ZclThermostatCluster.ATTR_SYSTEMMODE)) {
+                logger.trace("{}: Thermostat system mode not supported", endpoint.getIeeeAddress());
+                return null;
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.warn("{}: Exception discovering attributes in thermostat cluster", endpoint.getIeeeAddress(), e);
+        }
+
+        // TODO: Define the allowable states
```

Comment:
I guess this `TODO` is not intended to be resolved in this PR, but is rather a reminder for a potential future PR, correct?

Reply:
Correct. I didn't think this was possible when I wrote this, but I think your IAS PR shows me that there is an interface I didn't know about for providing the allowable channel values. It's something I need to look at...